scripts/prepare_sector_network.py

```diff
@@ -2789,6 +2791,261 @@ def add_electricity_distribution_grid(n, costs):
+
+    low_temp_buses = n.buses.loc[n.buses.carrier == 'low temperature heat 50-150C for industry'].index
+    medium_temp_buses = n.buses.loc[n.buses.carrier == 'medium temperature heat 150-250C for industry'].index
+
+    nodes_low = low_temp_buses.str.split(' ').str[0]
+    nodes_medium = medium_temp_buses.str.split(' ').str[0]
+
+    # Add carriers if not already present
+    carriers = [
+        "molten salt store",
+        "molten salt charger",
+        "molten salt discharger",
+        "solar heat for industrial processes",
+        "industrial heat pump high temperature",
+        "csp-tower",
+        "steam boiler gas cond",
+        "hot water boiler gas cond",
+        "hot water storage",
+    ]
+    for carrier in carriers:
+        if carrier not in n.carriers.index:
+            n.add("Carrier", name=carrier)
+
+    # 1. Low-temp molten salt store (Store)
+    n.madd(
+        "Bus",
+        nodes_medium + " molten salt store",
+        carrier="molten sand store"
+    )
+
+    n.madd(
+        "Store",
+        nodes_medium + " molten salt store",
+        bus=nodes_medium + " molten salt store",
+        carrier="molten salt store",
+        e_nom_extendable=True,
+        capital_cost=costs.at["low-temp molten salt store", "fixed"],
+        lifetime=costs.at["low-temp molten salt store", "lifetime"],
+        efficiency=costs.at["low-temp molten salt store", "efficiency"]
+    )
+
+    # 2. Low-temp molten salt discharger (Link)
+    # Assumes the discharger converts stored heat in the same bus to usable heat at the same bus
+    n.madd(
+        "Link", 
+        nodes_medium + " molten salt discharger",
+        bus0=nodes_medium + " molten salt store",
+        bus1=medium_temp_buses,
+        carrier="low-temp molten salt discharger",
+        p_nom_extendable=True,
+        capital_cost=costs.at["low-temp molten salt discharger", "fixed"],
+        lifetime=costs.at["low-temp molten salt discharger", "lifetime"],
+        efficiency=costs.at["low-temp molten salt discharger", "efficiency"],
+        p_min_pu=0.
+    )
+
+    # 3. Low-temp molten salt charger (Link)
+    logger.warning('Yet to get techno-economic data for molten sand charger')
+    n.madd(
+        "Link", 
+        nodes_medium + " molten salt charger",
+        bus0=medium_temp_buses,
+        bus1=nodes_medium + " molten salt store",
+        carrier="low-temp molten salt charger",
+        p_nom_extendable=True,
+        # capital_cost=costs.at["low-temp molten salt charger", "fixed"],
+        # lifetime=costs.at["low-temp molten salt charger", "lifetime"],
+        # efficiency=costs.at["low-temp molten salt charger", "efficiency"],
+        capital_cost=costs.at["low-temp molten salt discharger", "fixed"],
+        lifetime=costs.at["low-temp molten salt discharger", "lifetime"],
+        efficiency=costs.at["low-temp molten salt discharger", "efficiency"],
+        p_min_pu=0.
+    )
+
+    # 4. Solar heat for industrial processes (Generator)

```

Comment:
how does this differ from csp? is it a different type of collector?
the TODOs are better tagged with "TODO" it is better highlighted also from IDEs